packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -400,40 +396,39 @@ export class GutenbergEditorPage {
 	 * @param {boolean} visit Whether to then visit the page.
 	 * @returns {Promise<string>} The published URL.
 	 */
-	async publish( { visit = false }: { visit?: boolean } = {} ): Promise< string > {
+	async publish( {
+		update = false,
+		visit = false,
+	}: { update?: boolean; visit?: boolean } = {} ): Promise< URL > {
 		const frame = await this.getEditorFrame();
 
-		const initialPublishButton = `${ selectors.publishButton(
-			selectors.postToolbar
-		) }, ${ selectors.scheduleButton( selectors.postToolbar ) }`;
-		const secondaryPublishButton = `${ selectors.publishButton(
-			selectors.publishPanel
-		) }, ${ selectors.scheduleButton( selectors.publishPanel ) }`;
-
-		await frame.click( initialPublishButton );
-		await frame.click( secondaryPublishButton );
-		const publishedURL = await this.getPublishedURL();
-
-		if ( visit ) {
-			await this.visitPublishedPost( publishedURL );
+		if ( update ) {
```

Comment:
The `update` method has been deprecated and unified into `publish` method, since the contents of both the `publish` and `update` methods were nearly identical.

Reply:
Hmm, merging the update action into this method through a boolean parameter feels a little unintuitive to me semantically.  Calling `publish( { update: true } )` as the way to update a post (and the required way to to do it) seems like it might initially confuse callers. I think the confusion comes from mixing the syntax for actions that otherwise feeling similar (i.e., two actions are just auto-included by combined selector, but the third similar action is governed by a parameter).

I think we might benefit from taking this in one of two directions:
1. Make every distinct action (update, schedule, publish) its own public method. We can avoid code duplication by modularizing shared pieces into private class methods.
2. Just combine all three actions equally here without the boolean parameter. So this method will basically click the blue button in the top right of the editor no matter what it is, and handle whatever needs to be handled there.

What do you think?

Reply:
Hmm, good point. I will brood over this and evaluate both options.

Reply:
I've chosen Path 2 and unified the `publish` and `update` methods into one in 7df6ee16bfd28a4066cd09597dc5522f8a35f093.